File: zerojudge/d010/main.cpp
#include <bits/stdc++.h>
using namespace std;

int run(int N) {
    int sum = 0;
    vector <int> vec;
    for (int i = 1; i < N; i++){
        if (N == i){
            break;
        }
        if (N%i == 0){
            vec.push_back(i);
        }
    }

    for (int t = 0; t < vec.size(); t++){
        
        sum += vec[t];
    }
    if (N == sum) {
        cout<<"完全數"<<endl;
    }
    else if (N > sum){
        cout<<"虧數"<<endl;
    }
    else if (N < sum){
        cout<<"盈數"<<endl;
    }

    return 0;
}

int main(){
    int N;
    while (cin>>N) run(N);
    return 0;
}

File: zerojudge/d010/main.py
def run(N):
    l = []
    for i in range(1,N):
        if N % i == 0:
            l.append(i)
    if N == sum(l):
        print("完全數")
    elif N>sum(l):
        print("虧數")
    elif N<sum(l):
        print("盈數")   

def main():
    try:
        while True:
            N = int(input())
            run(N)
    except:
        return

main()